src/PhpSpec/Console/Application.php

```diff
@@ -145,16 +145,29 @@ protected function loadConfigurationFile(InputInterface $input, IndexedServiceCo
     {
         $config = $this->parseConfigurationFile($input);
 
+        $this->populateContainerParameters($container, $config);
+
         foreach ($config as $key => $val) {
             if ('extensions' === $key && is_array($val)) {
                 foreach ($val as $class => $extensionConfig) {
                     $this->loadExtension($container, $class, $extensionConfig ?: []);
                 }
             }
             elseif ('matchers' === $key && is_array($val)) {
-                $this->registerCustomMatchers($container, $val);
+                foreach ($val as $class) {
```

Comment:
Why have you replaced `registerCustomMatchers()` call with partial content of that method? You missed the code that ensured that passed matcher class is really a Matcher. Thats why the tests are red.

For me it makes sense to leave this block unchanged, as it seems to not be related to the PR.